subsys/net/ip/net_core.c

```diff
@@ -65,12 +66,66 @@ void net_analyze_stack(const char *name, const char *stack, size_t size)
 }
 #endif /* CONFIG_INIT_STACKS */
 
+#if defined(CONFIG_NET_SOCKETS_RAW)
+static inline void process_raw_data(struct net_pkt *pkt)
+{
+	struct net_context *cont;
+	struct net_pkt *raw_pkt;
+	int ret;
+
+	cont = net_context_find(AF_PACKET, SOCK_RAW, ETH_P_ALL);
+	if (!cont) {
+		return;
+	}
+
+	raw_pkt = net_pkt_clone(pkt, K_MSEC(100));
+	if (!raw_pkt) {
+		return;
+	}
+
+	net_pkt_set_family(raw_pkt, AF_PACKET);
+
+	ret = net_conn_input(ETH_P_ALL, raw_pkt);
+	if (ret != NET_OK) {
+		net_pkt_unref(raw_pkt);
+	}
+	/* TODO: It has to go through L2 layer, so that L2 will verifies
+	 * that packet intended to us or not and sets src and dst lladdr.
+	 * And L2 should not pullof L2 header in case of ETH_P_ALL proto.
+	 */
+#if 0
+	ret = net_if_recv_data(net_pkt_iface(raw_pkt), raw_pkt);
+	if (ret != NET_CONTINUE) {
+		if (ret == NET_DROP) {
+			NET_DBG("Packet %p discarded by L2", raw_pkt);
+			net_stats_update_processing_error(
+						net_pkt_iface(raw_pkt));
+		}
+
+		net_pkt_unref(raw_pkt);
+	}
+#endif
+}
+#else
+static inline void process_raw_data(struct net_pkt *pkt)
+{
+	;
+}
+#endif
+
 static inline enum net_verdict process_data(struct net_pkt *pkt,
 					    bool is_loopback)
 {
 	int ret;
 	bool locally_routed = false;
 
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_RAW)) {
+		/* TODO Loolback packets */
+		process_raw_data(pkt);
+
+		return NET_DROP;
```

Comment:
So as soon as CONFIG_NET_SOCKETS_RAW is enabled, all packet will be treated as raw stuff, and dropped?